### PR TITLE
Remove elastic agent from the beats repo

### PR DESCRIPTION
**Pull request by narph**
Remove elastic agent from the beats repo

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

### CODE REVIEW

**Review by kuisathaverat — APPROVED** [review]

File: .ci/packaging.groovy [hunk 1]
```diff
@@ -283,9 +281,7 @@ def pushCIDockerImages(Map args = [:]) {
     } else if (env?.BEATS_FOLDER?.endsWith('osquerybeat')) {
       tagAndPush(beatName: 'osquerybeat', arch: arch, variants: defaultVariants)
     } else if ("${env.BEATS_FOLDER}" == "packetbeat"){
-      tagAndPush(beatName: 'packetbeat', arch: arch, variants: defaultVariants)
-    } else if ("${env.BEATS_FOLDER}" == "x-pack/elastic-agent") {
-      tagAndPush(beatName: 'elastic-agent', arch: arch, variants: defaultVariants + completeVariant + cloudVariant)
+      tagAndPush(beatName: 'packetbeat', arch: arch)
```
Comment (v1v) [inline .ci/packaging.groovy:284]: Can you also remove the lines https://github.com/elastic/beats/pull/30559/files#diff-5edb5ab93fc95960129eecb2ee3d8763a9c5e02b56e5b80126e0c5a52296ee8fR270-R272 

Those lines contain the definition for completeVariant + cloudVariant which they are elastic-agent specific

File: Jenkinsfile [hunk 2]
```diff
@@ -452,8 +452,6 @@ def pushCIDockerImages(Map args = [:]) {
       tagAndPush(beatName: 'osquerybeat', arch: arch, variants: defaultVariants)
     } else if ("${beatsFolder}" == "packetbeat"){
       tagAndPush(beatName: 'packetbeat', arch: arch, variants: defaultVariants)
-    } else if ("${beatsFolder}" == "x-pack/elastic-agent") {
-      tagAndPush(beatName: 'elastic-agent', arch: arch, variants: defaultVariants + completeVariant + cloudVariant)
```
Comment (v1v) [inline Jenkinsfile:456]: ditto -> remove https://github.com/elastic/beats/pull/30559/files#diff-e6ffa5dc854b843b3ee3c3c28f8eae2f436c2df2b1ca299cca1fa5982e390cf8R440-R442 which contains they definition for `completeVariant + cloudVariant`

**Review by ph — APPROVED** [review]
LGTM, only comment from @v1v https://github.com/elastic/beats/pull/30559/files#r828093480

**Review by v1v — APPROVED** [review]

**Review by ph — APPROVED** [review]

**Comment by mdelapenya** [comment]
From e2e standpoint, there is one single failure that is completely not related: they are related to a regression, which will be eventually fixed in https://github.com/elastic/e2e-testing/pull/2246

👍

**Pull request merged**